src/module/actor/character/apps/attribute-builder/app.svelte

```diff
@@ -0,0 +1,943 @@
+    function buildBackgroundTooltip(
+        boostData: Record<string, { value: AttributeString[]; selected: AttributeString | null }>,
+    ): string | null {
+        const boostValues = Object.values(boostData);
+        if (
+            boostValues.length === 2 &&
+            boostValues[0].value.length === 2 &&
+            boostValues[1].value.length === ALL_ATTRIBUTES_COUNT
+        ) {
+            const choices = boostValues[0].value.map((b) => game.i18n.localize(CONFIG.PF2E.abilities[b]));
+            return game.i18n.format("PF2E.Actor.Character.AttributeBuilder.BackgroundBoostDescription", {
+                a: choices[0],
+                b: choices[1],
+            });
+        }
+        return null;
+    }
+
+    const ancestryBoosts = $derived.by(() => {
+        const ancestry = data.ancestry;
+        if (!ancestry) {
+            return null;
+        }
+
+        const [maxBoosts, selectedBoosts] = (() => {
+            const alternateAncestryBoosts = ancestry.system.alternateAncestryBoosts;
+            if (alternateAncestryBoosts) return [MAX_ALTERNATE_ANCESTRY_BOOSTS, alternateAncestryBoosts];
+
+            const baseBoosts = Object.values(ancestry.system.boosts);
+            const selectedBoosts = baseBoosts.map((b) => b.selected).filter((b): b is AttributeString => !!b);
+            const maxBoosts = baseBoosts.filter((b) => b.value.length > 0 || b.selected).length;
+            return [maxBoosts, selectedBoosts];
+        })();
+
+        const netBoosted = R.difference(data.build.boosts.ancestry, data.build.flaws.ancestry);
+        const remaining = maxBoosts - selectedBoosts.length;
+        const lockedBoosts = ancestry.system.alternateAncestryBoosts ? null : ancestry.lockedBoosts;
+        const lockedFlaws = ancestry.system.alternateAncestryBoosts ? null : ancestry.lockedFlaws;
+
+        const buttons = createButtonRecord(attributeList, (attribute) => {
+            const selected = selectedBoosts.includes(attribute);
+            return {
+                boost: {
+                    selected,
+                    locked: lockedBoosts?.includes(attribute),
+                    disabled: selected ? false : !remaining || netBoosted.includes(attribute),
+                },
+                flaw: lockedFlaws?.includes(attribute) ? { selected: true, locked: true } : undefined,
+            };
+        });
+
+        return {
+            buttons,
+            remaining,
+            labels: getBoostFlawLabels(ancestry.system.boosts),
+            flawLabels: getBoostFlawLabels(ancestry.system.flaws),
+            alternate: !!ancestry.system.alternateAncestryBoosts,
+        };
+    });
+
+    const legacyFlaws = $derived(data.ancestry?.system.voluntary?.boost !== undefined);
+
+    const voluntaryFlaws = $derived.by(() => {
+        const ancestry = data.ancestry;
+        if (!ancestry) {
+            return null;
+        }
```

Comment:
```suggestion
        if (!ancestry) return null;
```
tariffs on whitespace have sent prices through the roof :(